web_navbar_chameleon/models/res_company.py

```diff
@@ -0,0 +1,164 @@
+    def create(self, vals):
+        company_id = super(ResCompany, self).create(vals)
+        company_id.update_navbar_chameleon_scss()
+        return company_id
+
+    @api.multi
+    def unlink(self):
+        IrAttachmentObj = self.env["ir.attachment"]
+        for record in self:
+            custom_url = URL_SCSS_GEN % record.id
+            IrAttachmentObj.search([
+                ("url", '=', custom_url),
+            ]).unlink()
+        return super(ResCompany, self).unlink()
+
+    @api.onchange('logo')
+    def change_logo(self):
+        def n_rgb2hex(_r, _g, _b):
+            return '#%02x%02x%02x' % (int(255*_r), int(255*_g), int(255*_b))
+
+        if self.logo:
+            _r, _g, _b = self._get_image_mix_color_rgb(self.logo)
+            self.navbar_color_background = n_rgb2hex(_r, _g, _b)
+            # Make color 10% darker
+            _h, _l, _s = rgb_to_hls(_r, _g, _b)
+            _l = max(0, _l - 0.1)
+            _rd, _gd, _bd = hls_to_rgb(_h, _l, _s)
+            self.navbar_color_background_hover = n_rgb2hex(_rd, _gd, _bd)
+            # Calc. optimal text color (b/w)
+            # Grayscale human vision perception (rec601 values)
+            # https://www.itu.int/rec/R-REC-BT.601-7-201103-I/en
+            _a = 1 - (0.299 * _r + 0.587 * _g + 0.114 * _b)
+            self.navbar_color_text = '#000' if _a < 0.5 else '#fff'
+        else:
+            self.navbar_color_background = ''
+            self.navbar_color_background_hover = ''
+            self.navbar_color_text = ''
+
+    @api.onchange('navbar_color_background', 'navbar_color_background_hover',
+                  'navbar_color_text')
+    def change_navbar_color(self):
+        if self._origin.id:
+            datas = base64.b64encode(
+                self._generate_navbar_chameleon_scss().encode("utf-8"))
+            self._create_or_update_navbar_chameleon_scss_attachment(
+                self._origin.id, datas)
+
+    @api.multi
+    def _generate_navbar_chameleon_scss(self):
+        self.ensure_one()
+
+        return """
+            .o_main_navbar {
+              background-color: %(bg_color)s !important;
+              color: %(text_color)s !important;
+
+              > .o_menu_brand {
+                color: %(text_color)s !important;
+                &:hover, &:focus, &:active, &:focus:active {
+                  background-color: %(bghover_color)s !important;
+                }
+              }
+
+              .show {
+                .dropdown-toggle {
+                  background-color: %(bghover_color)s !important;
+                }
+              }
+
+              > ul {
+                > li {
+                  > a, > label {
+                    color: %(text_color)s !important;
+
+                    &:hover, &:focus, &:active, &:focus:active {
+                      background-color: %(bghover_color)s !important;
+                    }
+                  }
+                }
+              }
+            }
+        """ % {
```

Comment:
here I'd move all defaults to each respective field definition and do `self.SCSS_TEMPLATE.format(**self.company_colors)`